Sources/FoundationEssentials/WinSDK+Extensions.swift

```diff
@@ -288,4 +296,29 @@ internal func WIN32_FROM_HRESULT(_ hr: HRESULT) -> DWORD {
     return DWORD(hr)
 }
 
+internal func FillNullTerminatedWideStringBuffer(initialSize: DWORD, maxSize: DWORD, _ body: (UnsafeMutableBufferPointer<WCHAR>) throws -> DWORD) throws -> String {
+    var bufferCount = max(1, min(initialSize, maxSize))
+    while bufferCount < maxSize {
+        if let result = try withUnsafeTemporaryAllocation(of: WCHAR.self, capacity: Int(bufferCount), { buffer in
+            let count = try body(buffer)
+            switch count {
+            case 0:
+                throw Win32Error(GetLastError())
+            case 1..<DWORD(buffer.count):
+                guard let result = String.decodeCString(buffer.baseAddress!, as: UTF16.self)?.result else {
```

Comment:
Since `repairingInvalidCodeUnits` defaults to `true` here, this will never return `nil`. If the API contract is that this will always return valid UTF-16 upon success, should we just use `let result = String(decodingCString: buffer.baseAddress!, as: UTF16.self)` and remove the `guard`? That's what we do elsewhere in Foundation for APIs like this. If you do want to check for invalid UTF-16, then you'll need to pass `repairingInvalidCodeUnits: false` to instruct the function to return `nil` upon failure

Reply:
Opted for the path consistent with the rest of Foundation